go/api/base_client.go

```diff
@@ -69,14 +79,189 @@ func failureCallback(channelPtr unsafe.Pointer, cErrorMessage *C.char, cErrorTyp
 	resultChannel <- payload{value: nil, error: errors.GoError(uint32(cErrorType), msg)}
 }
 
+// RegisterClient registers a client in the registry using its pointer value
+func RegisterClient(client *baseClient, ptrValue uintptr) {
+	clientRegistryMu.Lock()
+	defer clientRegistryMu.Unlock()
+	clientRegistry[ptrValue] = client
+}
+
+// UnregisterClient removes a client from the registry
+func UnregisterClient(ptrValue uintptr) {
+	clientRegistryMu.Lock()
+	defer clientRegistryMu.Unlock()
+	delete(clientRegistry, ptrValue)
+}
+
+// GetClientByPtr gets a client from the registry by its pointer value
+func GetClientByPtr(ptrValue uintptr) *baseClient {
+	clientRegistryMu.RLock()
+	defer clientRegistryMu.RUnlock()
+	return clientRegistry[ptrValue]
+}
```

Comment:
These 3 functions replace the dispatcher, right?

Reply:
Yes. They allow us to take the coreClient ptr and get the associated Client.